src/containers/pages/FocusInvestigation/map/active/index.tsx

```diff
@@ -153,319 +152,344 @@ export const defaultMapSingleFIProps: MapSingleFIProps = {
   supersetService: supersetFetch,
 };
 
-/** Map View for Single Active Focus Investigation */
-class SingleActiveFIMap extends React.Component<
-  RouteComponentProps<RouteParams> & MapSingleFIProps,
-  FlexObject
-> {
-  public static defaultProps = defaultMapSingleFIProps;
-  constructor(props: RouteComponentProps<RouteParams> & MapSingleFIProps) {
-    super(props);
+const fetchData = async (
+  fetchGoalsActionCreator: typeof fetchGoals,
+  fetchJurisdictionsActionCreator: typeof fetchJurisdictions,
+  fetchPlansActionCreator: typeof fetchPlans,
+  fetchStructuresActionCreator: typeof setStructures,
+  fetchTasksActionCreator: typeof fetchTasks,
+  plan: Plan,
+  supersetService: typeof supersetFetch
+): Promise<void> => {
+  if (plan && plan.plan_id) {
+    /** define superset filter params for jurisdictions */
+    const jurisdictionsParams = superset.getFormData(SUPERSET_MAX_RECORDS, [
+      { comparator: plan.jurisdiction_id, operator: '==', subject: JURISDICTION_ID },
+    ]);
+    await supersetService(SUPERSET_JURISDICTIONS_SLICE, jurisdictionsParams)
+      .then((result: Jurisdiction[]) => fetchJurisdictionsActionCreator(result))
+      .catch(error => {
+        throw error;
+      });
+    /** define superset params for filtering by plan_id */
+    const supersetParams = superset.getFormData(SUPERSET_MAX_RECORDS, [
+      { comparator: plan.plan_id, operator: '==', subject: PLAN_ID },
+    ]);
+    /** define superset params for goals */
+    const goalsParams = superset.getFormData(
+      SUPERSET_MAX_RECORDS,
+      [{ comparator: plan.plan_id, operator: '==', subject: PLAN_ID }],
+      { action_prefix: true }
+    );
+    /** Implement Ad hoc Queries since jurisdictions have no plan_id */
+    await supersetService(SUPERSET_STRUCTURES_SLICE, jurisdictionsParams)
+      .then((structuresResults: Structure[]) => {
+        fetchStructuresActionCreator(structuresResults);
+      })
+      .catch(error => {
+        throw error;
+      });
+    await supersetService(SUPERSET_PLANS_SLICE, jurisdictionsParams)
+      .then((result2: Plan[]) => {
+        fetchPlansActionCreator(result2);
+      })
+      .catch(error => {
+        throw error;
+      });
+    await supersetService(SUPERSET_GOALS_SLICE, goalsParams)
+      .then((result3: Goal[]) => {
+        fetchGoalsActionCreator(result3);
+      })
+      .catch(error => {
+        throw error;
+      });
+    await supersetService(SUPERSET_TASKS_SLICE, supersetParams)
+      .then((result4: Task[]) => {
+        fetchTasksActionCreator(result4);
+      })
+      .catch(error => {
+        throw error;
+      });
+  }
+};
+
+const buildHandlers = () => {
```

Comment:
Ooh yeah let's export all of these for testability! Let's also include doc strings, and can we consider moving some of these into  an adjacent `helpers` file?